### PR TITLE
Adapt changes of `iota-client`

### PR DESCRIPTION
**Pull request by abdulmth**
# Description of change
Adapt the changes of the latest `iota-client` release.

## Type of change
- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] Enhancement (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Fix

### CODE REVIEW

File: identity_iota_core/src/client/iota_client.rs [hunk 1]
```diff
@@ -95,7 +96,7 @@ impl IotaClientExt for Client {
       .map_err(|err| Error::DIDUpdateError("delete_did_output: invalid block input", Some(Box::new(err))))?
       .with_outputs(vec![basic_output])
       .map_err(|err| Error::DIDUpdateError("delete_did_output: invalid block output", Some(Box::new(err))))?
-      .with_burning_allowed(true)
+      .with_burn(Burn::new().add_alias(alias_id))
```
Comment (abdulmth) [inline identity_iota_core/src/client/iota_client.rs:99]: @PhilippGackstatter could you please check this line? the alias output is everything we want to burn here, right?

Comment (PhilippGackstatter) [reply identity_iota_core/src/client/iota_client.rs:99]: That's right we only want to delete the alias, so what you changed looks correct.

**Review by PhilippGackstatter — APPROVED** [review]
LGTM!

**Review by olivereanderson — APPROVED** [review]
LGTM!

**Pull request merged**